connectors/airwatch/src/main/java/com/vmware/connectors/airwatch/AirWatchController.java

```diff
@@ -89,9 +91,10 @@ public AirWatchController(WebClient rest, CardTextAccessor cardTextAccessor,
         this.gbBaseUri = gbBaseUri;
     }
 
-    @GetMapping(path = "/discovery/metadata.json")
-    public ResponseEntity<String> getmetadata() {
-        return ResponseEntity.ok(connectorMetadata);
+    @GetMapping(path = METADATA_PATH)
+    public ResponseEntity<String> getmetadata(HttpServletRequest request) {
+        String requestUrl = request.getRequestURL().toString();
+        return ResponseEntity.ok(connectorMetadata.replace("CONNECTOR_HOST", requestUrl.split(METADATA_PATH)[0]));
```

Comment:
This isn't going to work.

You need to take into account:
* X-Forwarded-Host
* X-Forwarded-Port
* X-Forwarded-Proto

I'll dig up where this is happening in discovery.

Until then, you can test how this works by using curl (or postman) to interact with a connector's discovery:

```
curl -v \
     -H "Authorization: Bearer <vidm-token>" \
     -H "X-Forwarded-Proto: https" \
     -H "X-Forwarded-Host: www.my-host.com" \
     -H "X-Forwarded-Port: 8443" \
     "http://localhost:8080"
```

Reply:
You have to make sure you use UriComponentsBuilder.fromHttpRequest(HttpRequest) like we do in CommonUtils: https://github.com/vmware/connectors-workspace-one/blob/7c6bb345131f8a3c1e64eb67333a4cda90d26ffd/common/connectors-config/src/main/java/com/vmware/connectors/common/utils/CommonUtils.java#L41

That spring code is what is respecting the X-Forwarded headers.

Reply:
Thanks, I didn't think about 'X-Forwarded' at first.
Fixed it and tested from Postman.